### ISSUE TITLE
ZCAT_NSPEC in zpix-*.fits catalogs should report total / cumulative number of spectra

### ISSUE THREAD

**Issue by moustakas** [desihub/desispec#2085]
In the `zpix` summary files in a given production, e.g., `/global/cfs/cdirs/desi/spectro/redux/fuji/zcatalog/zpix-*.fits`, we report `ZCAT_NSPEC`, the "Number of coadded spectra for this TARGETID in this zcatalog"--
https://desidatamodel.readthedocs.io/en/latest/DESI_SPECTRO_REDUX/SPECPROD/zcatalog/zpix-SURVEY-PROGRAM.html#hdu1

However, for the healpix-coadded spectra, `ZCAT_NSPEC` is always one (1), e.g.,
```
set(fitsio.FITS('/global/cfs/cdirs/desi/spectro/redux/fuji/zcatalog/zpix-sv3-bright.fits')[1].read(columns='ZCAT_NSPEC'))
{1}
```

For the purposes of studying repeat spectra, this quantity should report the original / total number of spectra that went into the full healpix coadd. E.g.,
```
import fitsio
from astropy.table import Table
ztile = Table(fitsio.FITS('/global/cfs/cdirs/desi/spectro/redux/fuji/zcatalog/ztile-sv3-bright-cumulative.fits')['ZCATALOG'].read())
targetid = ztile[ztile['ZCAT_NSPEC'] == 8]['TARGETID'][0]

zpix = Table(fitsio.FITS('/global/cfs/cdirs/desi/spectro/redux/fuji/zcatalog/zpix-sv3-bright.fits')['ZCATALOG'].read())
exp = Table(fitsio.FITS('/global/cfs/cdirs/desi/spectro/redux/fuji/zcatalog/zpix-sv3-bright.fits')['EXP_FIBERMAP'].read())

exp[exp['TARGETID'] == targetid]
Out[29]:
<Table length=9>
     TARGETID      PRIORITY    SUBPRIORITY      NIGHT   EXPID      MJD       TILEID ... FIBER_Y DELTA_X DELTA_Y      FIBER_RA          FIBER_DEC      PSF_TO_FIBER_SPECFLUX
      int64         int32        float64        int32   int32    float64     int32  ... float64 float64 float64      float64            float64              float64
------------------ -------- ------------------ -------- ----- -------------- ------ ... ------- ------- ------- ------------------ ------------------ ---------------------
616093713713595510       -1 0.9998694785175377 20210414 84818 59319.37051261    205 ... -42.028   0.005     0.0 217.72728910590735  34.11516105752665    0.7906091530656002
616093713713595510       -1 0.9998694785175377 20210414 84819 59319.38248006    205 ... -42.028   0.005     0.0 217.72728910590735  34.11516105752665    0.7906091530656002
616093713713595510       -1 0.9998694785175377 20210418 85342 59323.30397978    207 ... -98.653  -0.004  -0.002 217.72724450009477  34.11516927769747    0.7907461014117874
616093713713595510       -1 0.9998694785175377 20210420 85633 59325.33545773    208 ... -52.207    0.01  -0.008 217.72731397903846  34.11519385659506    0.7917210982888288
616093713713595510       -1 0.9998694785175377 20210430 86747 59335.35124006    210 ...  -83.71  -0.008   0.004 217.72722461766514  34.11514477266212    0.7911653768611281
616093713713595510       -1 0.9998694785175377 20210405 83535 59310.23480542    201 ... -92.554  -0.003   0.003 217.72724946690633  34.11514883714701    0.7907289962706423
616093713713595510       -1 0.9998694785175377 20210416 85094 59321.36555168    206 ... -79.123   0.018  -0.002  217.7273533989625  34.11516922623964    0.7928687986030847
616093713713595510       -1 0.9998694785175377 20210428 86503 59333.30986707    209 ... -62.523  -0.004  -0.003 217.72724446784147 34.115173386467866    0.7906881005528615
616093713713595510       -1 0.9998694785175377 20210501 86841 59336.22817923    211 ... -40.589   0.003   0.002 217.72727917586306  34.11515286247112    0.7905523169234739

zpix[zpix['TARGETID'] == targetid]['ZCAT_NSPEC']
<Column name='ZCAT_NSPEC' dtype='int16' length=1>
1
```

**Comment by sbailey** [comment]
The datamodel description for `ZCAT_NSPEC` "Number of coadded spectra for this TARGETID in this zcatalog" is confusing.

That variable is *not* the number of spectra which were coadded to create this entry in the catalog (that is COADD_NUMEXP), but rather it is trying to describe cases like the zall catalogs or the pernight ztile catalogs, where the same TARGETID can appear multiple times, and it is counting how many times that it.  In the case of the zpix catalogs, they are already grouped by survey and program and thus each TARGETID appears only once thus ZCAT_NSPEC=1.  A clearer description might be

> Number of times this TARGETID appears in this catalog"

@moustakas does this clarify the distinction between `ZCAT_NSPEC` and `COADD_NUMEXP`, and/or do you have alternate wording to suggest?

**Comment by sbailey** [comment]
Yes, the "fix" would be at a documentation level in desidatamodel, but I wanted to respond with the initial rewording suggestion here in the ticket where the question was raised.

FWIW, I was also initially confused by ZCAT_NSPEC and didn't (re-)realize what it really meant until digging into the code looking for a bug, but then I understood what the datamodel wording meant, and hopefully we can clarify that.

**Comment by weaverba137** [comment]
@sbailey, are any additinal *code* changes needed in desispec needed to close this, or has this transitioned to more of a desidatamodel ticket?

**Comment by sbailey** [comment]
Yes, this is effectively now a desidatamodel ticket.  It is probably faster to address it and close rather than figure out how to transition the ticket to a different repo.  If/when someone updates the text, please assign this ticket to yourself so that we don't accidentally have two people editing desidatamodel for the same entry at the same time.

**Comment by weaverba137** [comment]
In order to clean this up, what is the new definition needed for `ZCAT_NSPEC`? Is it just

> "Number of times this TARGETID appears in this catalog"

or something more?

**Comment by sbailey** [comment]
> In order to clean this up, what is the new definition needed for ZCAT_NSPEC? Is it just

>> "Number of times this TARGETID appears in this catalog"

> or something more?

Yes, pleased update the description in desidatamodel (and directly commit, no need to re-review it there), and then close this ticket.  Thanks.

**Comment by weaverba137** [comment]
Closed by desihub/desidatamodel#187.

**Comment by weaverba137** [comment]
@sbailey, Why was I assigned a closed ticket?
